frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-topology/PipelineTaskNode.tsx

```diff
@@ -0,0 +1,136 @@
+import * as React from 'react';
+import { PopoverProps } from '@patternfly/react-core';
+import {
+  DEFAULT_LAYER,
+  DEFAULT_WHEN_OFFSET,
+  Layer,
+  Node,
+  ScaleDetailsLevel,
+  TaskNode,
+  TOP_LAYER,
+  useDetailsLevel,
+  useHover,
+  WhenDecorator,
+  WithContextMenuProps,
+  WithSelectionProps,
+} from '@patternfly/react-topology';
+import { observer } from 'mobx-react';
+import { Link } from 'react-router-dom';
+import { resourcePathFromModel } from '@console/internal/components/utils';
+import { PipelineRunModel } from '../../../models';
+import { ComputedStatus } from '../../../types';
+import {
+  createStepStatus,
+  StepStatus,
+} from '../detail-page-tabs/pipeline-details/pipeline-step-utils';
+import { PipelineVisualizationStepList } from '../detail-page-tabs/pipeline-details/PipelineVisualizationStepList';
+
+type PipelineTaskNodeProps = {
+  element: Node;
+} & WithContextMenuProps &
+  WithSelectionProps;
+
+const PipelineTaskNode: React.FunctionComponent<PipelineTaskNodeProps> = ({
+  element,
+  onContextMenu,
+  contextMenuOpen,
+  ...rest
+}) => {
+  const data = element.getData();
+  const [hover, hoverRef] = useHover();
+  const detailsLevel = useDetailsLevel();
+  const stepList = data?.task?.status?.steps || [];
+  const stepStatusList: StepStatus[] = stepList.map((step) =>
+    createStepStatus(step, data?.task?.status),
+  );
+  const { pipelineRun } = data;
+  const succeededStepsCount =
+    stepStatusList.filter(({ status }) => status === ComputedStatus.Succeeded)?.length || 0;
```

Comment:
Nit:

`stepStatusList`  is always defined, and if not this will crash without optional changing, so we don't the optional check and fallback to zero here:

```suggestion
  const succeededStepsCount = stepStatusList.filter(({ status }) => status === ComputedStatus.Succeeded).length;
```